sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -729,6 +732,13 @@ public void validate() {
     private Instant curTimestamp;
     private Iterator<PartitionState> curBatch = Collections.emptyIterator();
 
+    private final Counter messagesConsumed = Metrics.counter("io", "messagesConsumed");
+    private final Counter bytesConsumed = Metrics.counter("io", "bytesConsumed");
```

Comment:
So, regarding `elementsRead` that's true but regarding other metrics such as `backlog` it may not be true, for example, backlog can be represented by things other than bytes, which is the only thing exposed by the source API.
The idea was instead of exposing things in the source API and then having runners report these as metrics, the sources would report metrics directly from sources, saving this work from the runner, and reporting things the runner does not have access to.